### PR TITLE
Add free-pascal demangler (fpc 2.6.x to 3.2.x)

### PR DESCRIPTION
**Pull request by wargio**
Adds support for the fpc demangler

Fix (partially) #16

### CODE REVIEW

File: test/test_pascal.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "demangling_unit.h"
+
+mu_demangle(0, pascal, "OUTPUT_$$_SQUARE$SMALLINT$$SMALLINT", "unit output square(var a0: smallint) smallint");
+mu_demangle(1, pascal, "OUTPUT_$$_init", "unit output init()");
+mu_demangle(2, pascal, "OUTPUT$_$MYOBJECT_$__$$_INIT$$QWORDBOOL", "unit output myobject.init() qwordbool");
+mu_demangle(3, pascal, "OUTPUT$_$MYOBJECT_$__$$_MYMETHOD", "unit output myobject.mymethod()");
+mu_demangle(4, pascal, "OUTPUT_$$_MYFUNC$$POINTER", "unit output myfunc() pointer");
+mu_demangle(5, pascal, "OUTPUT_$$_MYPROCEDURE$SMALLINT$LONGINT$PCHAR", "unit output myprocedure(var a0: smallint; a1: longint; a2: pchar)");
+mu_demangle(6, pascal, "OUTPUT_$$_MYFUNC2$SMALLINT$LONGINT$PCHAR$$POINTER", "unit output myfunc2(var a0: smallint; a1: longint; a2: pchar) pointer");
```
Comment (wargio) [inline test/test_pascal.c:12]: kinda unsure if is better to simplify this to `myfunc2(smallint,longint,pchar)` like fpc does.

Comment (XVilka) [reply test/test_pascal.c:12]: Both variants are fine, I think.

File: src/pascal/pascal.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
+// SPDX-FileCopyrightText: 2021 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+#include "demangler_util.h"
```
Comment (XVilka) [inline src/pascal/pascal.c:3]: Please keep one empty line between SPDX header and includes

**Review by XVilka — APPROVED** [review]
Looks good.

**Pull request merged**